### DISCUSSION TITLE
How to edit each paragraph node correctly?

### DISCUSSION THREAD

**Discussion by SF-Simon** [ueberdosis/tiptap#2259] [General Questions] [answered]
We want to check the characters of each paragraph. If there are qualified paragraphs in it, we need to modify the text of the node, and refresh the view of the editor after batch checking.

First, we pass the `view.docView.children` traverse to get each paragraph (not a node), and then we can use `node.text` get the corresponding value, and modify and change it.

But next, we don't know how to feed back to the editor to tell it that the content of the node has been modified and that the view needs to be refreshed.

thx.so much

**Comment by philippkuehn** [comment, marked as answer]
Mutating the dom directly is not supported by ProseMirror. You have to iterate over `state.doc.descendants` and create a transaction to change the state.

**Reply by SF-Simon** [reply]
Can you provide a demonstration code? I don't know whether we understand it wrong or the code is wrong and doesn't meet expectations. thank you.

**Reply by philippkuehn** [reply]
I think this search & replace issue can be a good starting point to you https://github.com/ueberdosis/tiptap/issues/1891
